janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -227,114 +159,71 @@ public CQLStoreManager(final Configuration configuration) throws BackendExceptio
-                    .build()));
+            configLoaderBuilder
+                .withClass(DefaultDriverOption.AUTH_PROVIDER_CLASS, PlainTextAuthProvider.class)
+                .withString(DefaultDriverOption.AUTH_PROVIDER_USER_NAME, configuration.get(AUTH_USERNAME))
+                .withString(DefaultDriverOption.AUTH_PROVIDER_PASSWORD, configuration.get(AUTH_PASSWORD));
         }
 
         if (configuration.get(SSL_ENABLED)) {
-            try {
-                KeyManager[] keyManagers = null;
-                if(configuration.get(SSL_CLIENT_AUTHENTICATION_ENABLED)) {
-                    try (final FileInputStream keyStoreStream = new FileInputStream(configuration.get(SSL_KEYSTORE_LOCATION))) {
-                        final KeyStore keystore = KeyStore.getInstance("jks");
-                        keystore.load(keyStoreStream, configuration.get(SSL_KEYSTORE_STORE_PASSWORD).toCharArray());
-                        final KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-                        keyManagerFactory.init(keystore, configuration.get(SSL_KEYSTORE_KEY_PASSWORD).toCharArray());
-                        keyManagers = keyManagerFactory.getKeyManagers();
-                    }
-                }
-                final TrustManager[] trustManagers;
-                try (final FileInputStream keyStoreStream = new FileInputStream(configuration.get(SSL_TRUSTSTORE_LOCATION))) {
-                    final KeyStore keystore = KeyStore.getInstance("jks");
-                    keystore.load(keyStoreStream, configuration.get(SSL_TRUSTSTORE_PASSWORD).toCharArray());
-                    final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-                    trustManagerFactory.init(keystore);
-                    trustManagers = trustManagerFactory.getTrustManagers();
-                }
-
-                final SSLContext sslContext = SSLContext.getInstance("TLS");
-                sslContext.init(keyManagers, trustManagers, null);
-
-                final JdkSSLOptions sslOptions = JdkSSLOptions.builder()
-                        .withSSLContext(sslContext)
-                        .build();
-                builder.withSSL(sslOptions);
-
-            } catch (UnrecoverableKeyException | NoSuchAlgorithmException | CertificateException | IOException | KeyStoreException | KeyManagementException e) {
-                throw new PermanentBackendException("Error initialising SSL connection properties", e);
-            }
+            configLoaderBuilder
+                .withClass(DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS, DefaultSslEngineFactory.class)
+                .withString(DefaultDriverOption.SSL_KEYSTORE_PATH, configuration.get(SSL_KEYSTORE_LOCATION))
+                .withString(DefaultDriverOption.SSL_KEYSTORE_PASSWORD, configuration.get(SSL_KEYSTORE_KEY_PASSWORD))
```

Comment:
@porunov Build is  failing on `cassandra3-murmur-ssl` with:
```
java.lang.IllegalArgumentException: Invalid configuration value for [root.storage.cql.ssl.keystore.location]:
```
This is because we're setting the key store options when client auth has not been enabled. So these options need moved out to something like:
```java
            if(configuration.get(SSL_CLIENT_AUTHENTICATION_ENABLED)) {
            	configLoaderBuilder.withString(DefaultDriverOption.SSL_KEYSTORE_PATH, configuration.get(SSL_KEYSTORE_LOCATION))
                                   .withString(DefaultDriverOption.SSL_KEYSTORE_PASSWORD, configuration.get(SSL_KEYSTORE_KEY_PASSWORD));
            }
```